Switch to platform widgets

Hamitix opened the pull request.

Hamitix committed:
Added 'platform specific widgets' library & new wrapper for specific widgets

Hamitix committed:
Removed "beach.jpg" in asset

Hamitix committed:
Updated theme's files

Hamitix committed:
Added platform widgets to the Home, BMI , TreeList, TreeDetail & Settings screens

dpocheronOnePoint commented on assets/translations/es.json:
```diff
@@ -35,6 +35,7 @@
 
     "bmi_screen" : {
         "weight": "peso",
```
Height is capitalize but not Weight

dpocheronOnePoint commented on assets/translations/es.json:
```diff
@@ -35,6 +35,7 @@
 
     "bmi_screen" : {
         "weight": "peso",
+        "height": "Altura",
```
Height is capitalize but not Weight

dpocheronOnePoint replied:
Height is capitalize but not Weight

dpocheronOnePoint commented on assets/translations/fr.json:
```diff
@@ -35,6 +35,7 @@
 
     "bmi_screen" : {
         "weight": "poids",
+        "height": "Taille",
```
Height is capitalize but not Weight

Hamitix committed:
Fixed some translations

Hamitix committed:
Added specific toggle buttons for each platform

dpocheronOnePoint commented on lib/shared/platform_text_wrapper.dart:
```diff
@@ -0,0 +1,21 @@
+          );
+
+TextStyle _defaultIosStyle = const TextStyle(
+    color: Colors.white, decoration: TextDecoration.none, fontSize: 14);
```
Why don't use PlatformText on Android?

dpocheronOnePoint commented on lib/translations/codegen_loader.g.dart:
```diff
@@ -184,7 +190,9 @@ static const Map<String,dynamic> es = {
+    "throw_test_exception": "Lanzar una excepción de prueba"
   },
   "langage": {
     "english_langage": "Inglés",
```
You need to run flutter pub run easy_localization:generate -S "assets/translations" -O "lib/translations" to update keys

ZMontassar commented on lib/screens/settings/settings_view_model.dart:
```diff
@@ -98,53 +98,97 @@ abstract class SettingsViewModelBase with Store, ViewModel {
+          await context.setLocale(Locale(languageCode)),
           // ignore: use_build_context_synchronously
-          Phoenix.rebirth(context);
+          Phoenix.rebirth(context)
```
I don't think we need `Phoenix.rebirth` because we only use it when we want to rebuild your application.
and when I click on a button to switch to another language I notice that the screen goes black for a second and then it shows up again and I think that it `Phoenix` that does this.

Hamitix committed:
Clean some codes on the map related files

Hamitix committed:
Fixed some translations

Hamitix committed:
- Added a "cupertinoTabChildBuilder" to the platform widgets in the home screen to fixed some bugs in iOS with TabView
- Changed the implentation of the appBar in the "TabView" => Each screen has its specific appBar

ZMontassar commented on lib/ressources/theme/green/theme_green_light.dart:
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
 
-Color lightGreen = const Color(0xff83f28f);
+Color lightGreen = const Color(0xff65793C);
```
I think we can create a Color class, is that we will be able to access the colors from this class anywhere and change them when we need, Like : 

`import 'package:flutter/material.dart';
class ColorSelect {
  static final lightGreen = const Color(0xff65793C);
}`

Hamitix committed:
Renamed the filename of the "row_tree_info" component

Hamitix committed:
Require the context when using the "text_platform" custom widget

Hamitix committed:
Added a new platform wrapper for separated ListView using the "CupertinoListTile" library

Hamitix committed:
Added a "ThemeColors" file with static colors

Hamitix committed:
FIxed a color name in the "ThemeColor" file

Hamitix committed:
Refactor "TreeDetailScreen" to Stateless Widget

Hamitix committed:
Renaming of the imports of the custom themes
